query/src/interpreters/interpreter_table_show_create.rs

```diff
@@ -59,26 +59,37 @@ impl Interpreter for ShowCreateTableInterpreter {
         let name = table.name();
         let engine = table.engine();
         let schema = table.schema();
+        let field_comments = table.field_comments();
+        let n_fields = schema.fields().len();
 
         let mut table_create_sql = format!("CREATE TABLE `{}` (\n", name);
 
         // Append columns.
         {
             let mut columns = vec![];
-            for field in schema.fields().iter() {
+            for (idx, field) in schema.fields().iter().enumerate() {
                 let default_expr = match field.default_expr() {
                     Some(expr) => {
                         let expression = PlanParser::parse_expr(expr)?;
                         format!(" DEFAULT {}", expression.column_name())
                     }
                     None => "".to_string(),
                 };
+                // compatibility: creating table in the old planner will not have `fields_comments`
+                let comment = if field_comments.len() == n_fields && !field_comments[idx].is_empty()
+                {
+                    format!(" COMMENT '{}'", &field_comments[idx])
```

Comment:
Does the `field_comments[idx]` need to be escaped?

Reply:
E.g. when there is a single quote `'` in the comment

Reply:
Oh, I didn't consider this. I'll take a look.

Reply:
```sql
create table t (a int comment 'aaa\'bbb\'ccc');
show create table t;
```

MySQL's result:

```
+-------+--------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                         |
+-------+--------------------------------------------------------------------------------------------------------------------------------------+
| t     | CREATE TABLE `t` (
  `a` int DEFAULT NULL COMMENT 'aaa''bbb''ccc'
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci |
+-------+--------------------------------------------------------------------------------------------------------------------------------------+
```

MySQL split the string.

This PR's result:

```
+-------+------------------------------------------------------------------+
| Table | Create Table                                                     |
+-------+------------------------------------------------------------------+
| t     | CREATE TABLE `t` (
  `a` INT COMMENT 'aaa'bbb'ccc'
) ENGINE=FUSE |
+-------+------------------------------------------------------------------+
```

Should we be the same with MySQL's behavior?

Reply:
It's simple to just replace `'` with `\'` ?

Reply:
There will not be `'` in the comment, only `\'`. Because we cannot write `'aaa'bbb'ccc'` in SQL, but only can write `'aaa\'bbb\'ccc'`. The problem now is if we should follow MySQL style, or just keep the style above.

Reply:
I think we do not need to follow MySQL style,  `'aaa''bbb''ccc'` is hard to read.

Reply:
But we should be able to create a new table with this format SQL.

Reply:
You means `create table t (a int comment 'aaa''bbb''ccc');` ? This will fail during the parse phase.